lib/active_merchant/billing/gateways/braintree_blue.rb

```diff
@@ -171,8 +171,17 @@ def unstore(customer_vault_id, options = {})
       end
       alias_method :delete, :unstore
 
+      def network_tokenization_supported
+        result = Braintree::MerchantGateway.new(@braintree_gateway).provision_raw_apple_pay
+        if result.success?
+          result.supported_networks.map { |brand| format_card_brand(brand) }
+        else
+          []
+        end
+      end
+
       def supports_network_tokenization?
-        true
+        network_tokenization_supported.present?
```

Comment:
no since  `network_tokenization_supported` will always return an array of the supported card brands